core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -195,6 +198,61 @@ class ControllerApis(val requestChannel: RequestChannel,
       requestThrottleMs => createResponseCallback(requestThrottleMs))
   }
 
+  def handleDeleteTopics(request: RequestChannel.Request): Unit = {
+    if (!config.deleteTopicEnable) {
+      if (request.context.apiVersion() < 3) {
+        throw new InvalidRequestException("Topic deletion is disabled.")
+      } else {
+        throw new TopicDeletionDisabledException()
+      }
+    }
+    val deleteTopicsRequest = request.body[DeleteTopicsRequest]
+    val nameToId = new mutable.HashMap[String, Uuid]
+    deleteTopicsRequest.data().topicNames().iterator().asScala.foreach {
+      name => nameToId.put(name, Uuid.ZERO_UUID)
+    }
+    deleteTopicsRequest.data().topics().iterator().asScala.foreach {
+      nameAndId => nameToId.put(nameAndId.name(), nameAndId.topicId())
+    }
+    val (describable, deletable)  = {
+      if (authHelper.authorize(request.context, DELETE, CLUSTER, CLUSTER_NAME)) {
+        (nameToId.keySet, nameToId.keySet)
+      } else {
+        val authorizedDescribeTopics: Set[String] = authHelper.filterByAuthorized(
+          request.context, DESCRIBE, TOPIC, nameToId.keys)(n => n)
+        val authorizedDeleteTopics: Set[String] = authHelper.filterByAuthorized(
+          request.context, DELETE, TOPIC, nameToId.keys)(n => n)
+        (authorizedDescribeTopics, authorizedDeleteTopics)
+      }
+    }
+    def sendResponse(response: DeleteTopicsResponseData): Unit = {
+      nameToId.keysIterator.foreach {
+        name => if (!deletable.contains(name)) {
+          val result = if (describable.contains(name)) {
+            new DeletableTopicResult().setName(name).setErrorCode(TOPIC_AUTHORIZATION_FAILED.code)
+          } else {
+            new DeletableTopicResult().setName(name).setErrorCode(UNKNOWN_TOPIC_OR_PARTITION.code)
+          }
+          response.responses().add(result)
+        }
+      }
+      requestHelper.sendResponseMaybeThrottle(request, throttleTimeMs => {
+        response.setThrottleTimeMs(throttleTimeMs)
+        new DeleteTopicsResponse(response)
+      })
+    }
+   val future = controller.deleteTopics(
+     nameToId.view.filterKeys(deletable.contains(_)).toMap.asJava)
```

Comment:
It can be simplified to `nameToId.view.filterKeys(deletable.contains).toMap.asJava)`

Reply:
I would use `filter` and remove `view` and `toMap`.

Reply:
this has changed